assets/src/edit-story/components/panels/design/animation/effectChooserDropdown/effectChooserDropdown.js

```diff
@@ -0,0 +1,165 @@
+
+export default function EffectChooserDropdown({
+  onAnimationSelected,
+  onNoEffectSelected,
+  isBackgroundEffects = false,
+  selectedEffectType,
+  disabledTypeOptionsMap,
+  direction,
+}) {
+  const { enableExperimentalAnimationEffects } = useFeatures();
+
+  const selectedValue = useMemo(
+    () => getDirectionalEffect(selectedEffectType, direction),
+    [selectedEffectType, direction]
+  );
+  // BACKGROUND EFFECT SET UP
+  const getDisabledBackgroundEffects = useCallback(() => {
+    const disabledDirectionalEffects = Object.entries(disabledTypeOptionsMap)
+      .map(([effect, val]) => [effect, val.options])
+      .reduce(
+        (directionalEffects, [effect, directions]) => [
+          ...directionalEffects,
+          ...(directions || []).map((dir) => getDirectionalEffect(effect, dir)),
+        ],
+        []
+      );
+    return Object.keys(backgroundEffectOptions).filter((directionalEffect) =>
+      disabledDirectionalEffects.includes(directionalEffect)
+    );
+  }, [disabledTypeOptionsMap]);
+
+  const disabledBackgroundEffects = getDisabledBackgroundEffects();
+
+  // Set up dropdown options by effect type (background vs foreground)
+  const expandedPlacement = isBackgroundEffects
+    ? PLACEMENT.BOTTOM
+    : PLACEMENT.TOP;
+  const animationOptionsObject = isBackgroundEffects
+    ? backgroundEffectOptions
+    : foregroundEffectOptions;
+
+  // remove experiments if needed
+  const availableAnimationOptions = useMemo(() => {
+    return enableExperimentalAnimationEffects
+      ? Object.values(animationOptionsObject)
+      : Object.values(animationOptionsObject).filter(({ value }) => {
+          return experimentalEffects.indexOf(value) === -1;
+        });
+  }, [animationOptionsObject, enableExperimentalAnimationEffects]);
+
+  const assembledOptions = useMemo(() => {
+    return availableAnimationOptions?.map((option) => {
+      const isDisabled =
+        isBackgroundEffects && disabledBackgroundEffects.includes(option.value);
+      return {
+        ...option,
+        disabled: isDisabled,
+        tooltip:
+          isDisabled &&
+          disabledTypeOptionsMap[option.animation?.value]?.tooltip,
+      };
+    });
+  }, [
+    availableAnimationOptions,
+    isBackgroundEffects,
+    disabledBackgroundEffects,
+    disabledTypeOptionsMap,
+  ]);
+
+  const handleSelect = useCallback(
+    (_, value) => {
+      if (value === NO_ANIMATION) {
+        onNoEffectSelected();
+      }
+
+      const selectedAnimation = animationOptionsObject[value].animation;
+
+      onAnimationSelected({
+        animation: selectedAnimation.value,
+        panDir: selectedAnimation?.panDirection,
+        zoomDirection: selectedAnimation?.zoomDirection,
+        flyInDir: selectedAnimation?.flyInDirection,
+        rotateInDir: selectedAnimation?.rotateInDirection,
+        whooshInDir: selectedAnimation?.whooshInDirection,
+        scaleDirection: selectedAnimation?.scaleDirection,
```

Comment:
I evened out the naming internal here with `dropDownConstants` but didn't want to rename things outside of this area to avoid scope creep.